src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -39,6 +39,7 @@ private sealed class Http2Stream : IValueTaskSource, IHttpHeadersHandler, IHttpT
             private int _pendingWindowUpdate;
             private CreditWaiter? _creditWaiter;
             private int _availableCredit;
+            private object _creditSyncObject => new object(); // split from SyncObject to mitigate deadlock in OnWindowUpdate.
```

Comment:
Despite the naming, you've written a property here rather than a field, so this is not doing what you want: every access to _creditSyncObject is creating a new object.

Reply:
:facepalm: fixing now